src/main/java/com/greedobank/reports/service/ReportService.java

```diff
@@ -0,0 +1,138 @@
+package com.greedobank.reports.service;
+
+import com.greedobank.reports.dao.NewsDAO;
+import com.greedobank.reports.model.CustomCellStyle;
+import com.greedobank.reports.model.News;
+import com.greedobank.reports.utils.StylesGenerator;
+import org.apache.poi.ss.usermodel.Cell;
+import org.apache.poi.ss.usermodel.CellStyle;
+import org.apache.poi.ss.usermodel.Row;
+import org.apache.poi.ss.usermodel.Sheet;
+import org.apache.poi.ss.usermodel.Workbook;
+import org.apache.poi.xssf.usermodel.XSSFSheet;
+import org.apache.poi.xssf.usermodel.XSSFWorkbook;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.time.OffsetDateTime;
+import java.time.temporal.ChronoUnit;
+import java.util.List;
+import java.util.Map;
+
+@Service
+public class ReportService {
+    private final StylesGenerator stylesGenerator;
+    private final NewsDAO newsDAO;
+    private final static int DESCRIPTION_INDEX = 4;
+    private final static int NUMBERS_OF_COLUMNS = 9;
+
+
+    @Autowired
+    public ReportService(StylesGenerator stylesGenerator, NewsDAO newsDAO) {
+        this.stylesGenerator = stylesGenerator;
+        this.newsDAO = newsDAO;
+    }
+
+    public byte[] generateXlsxReport() throws IOException {
+        var wb = new XSSFWorkbook();
+
+        generateReport(wb);
+        try (var out = new ByteArrayOutputStream()) {
+            wb.write(out);
+            return out.toByteArray();
+        }
+    }
+
+    private Workbook generateReport(XSSFWorkbook wb) {
+        var styles = stylesGenerator.prepareStyles(wb);
```

Comment:
why did you decide to mix `var`s and "types"?

Reply:
and do we need to execute `stylesGenerator.prepareStyles` every time, but not store these styles as a bean/other L1 cache entity?